src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/updatelistingrequirements/UpdateListingRequirementsHandler.java

```diff
@@ -0,0 +1,210 @@
+            .setListItems(interpreterLanguage);
+    }
+
+    public void generateReservedToJudgeFields(OverrideFields overrideFields) {
+        if (isNull(overrideFields.getReservedToJudge())
+            || isNull(overrideFields.getReservedToJudge().getReservedMember())) {
+            overrideFields.setReservedToJudge(ReservedToMember.builder()
+                .reservedMember(new DynamicList(null, null))
+                .build());
+        }
+
+        List<DynamicListItem> reservedMembers = generateReservedMembers();
+
+        overrideFields.getReservedToJudge().getReservedMember().setListItems(reservedMembers);
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
+
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        return response;
+    }
+
+    @NotNull
+    public List<DynamicListItem> generateInterpreterLanguage() {
+
+        return getLanguages().stream()
+            .map(this::getLanguageDynamicListItem)
+            .collect(Collectors.toList());
+    }
+
+    @NotNull
+    public DynamicListItem getLanguageDynamicListItem(Language language) {
+        String reference = language.getReference();
+        String name = language.getNameEn();
+
+        if (nonNull(language.getDialectReference())) {
+            reference = String.format("%s-%s", language.getReference(), language.getDialectReference());
+            name = language.getDialectEn();
+        }
+        return new DynamicListItem(reference, name);
+    }
+
+    public List<Language> getLanguages() {
+        List<Language> signLanguages = signLanguagesService.getSignLanguages();
+        List<Language> verbalLanguages = verbalLanguagesService.getVerbalLanguages();
+
+        return Stream.concat(signLanguages.stream(), verbalLanguages.stream())
+            .collect(Collectors.toList());
+    }
+
+    public List<DynamicListItem> generateReservedMembers() {
+        JudicialRefDataUsersRequest request = JudicialRefDataUsersRequest.builder()
+            .ccdServiceName(SERVICE_NAME)
+            .build();
+
+        JudicialRefDataUsersResponse refDataResponse = judicialRefData.getJudicialUsers(
+            idamService.getIdamTokens().getIdamOauth2Token(),
+            idamService.getIdamTokens().getServiceAuthorization(),
+            request);
+
+        return Optional.ofNullable(refDataResponse.getJudicialUsers())
+            .orElse(Collections.emptyList())
+            .stream()
+            .filter(Objects::nonNull)
+            .filter(judicialUser -> isNotEmpty(judicialUser.getAppointments()))

```

Comment:
Could the appointments be null?

Reply:
Sonar didn't like it not being null checked